### ISSUE TITLE
runTimer not working on 2 specific songs???

### ISSUE THREAD

**Issue by FreakyOnAFriday** [ShadowMario/FNF-PsychEngine#10185]
### runTimer not working on two specific songs???

So I have a custom notetype that, whenever you miss the note, is supposed to have a runTimer of 'bleed' for 20 seconds. It works on the first song I've made, but for the second and third songs the runTimer glitches out and doesn't work, and I don't loose any health/bleed. I've checked all my code and I didn't find anything that could possibly be a cause of this. (At first I thought it was something I did) Does anyone know what could possibly be a cause of this and if so how to fix it? I don't want to have to rechart the entire song. I also made a test song and the runTimer worked correctly on it. So needless to say I'm mildly confused.

### Are you modding a build from source or with Lua?

Lua

### What is your build target?

Windows

### Did you edit anything in this build? If so, mention or summarize your changes.

no

**Comment by frantastic24** [comment]
you didn't post the script

**Comment by FreakyOnAFriday** [comment]
> you didn't post the script

My bad.


`function onCreate()
	--Iterate over all notes
	for i = 0, getProperty('unspawnNotes.length')-1 do
		if getPropertyFromGroup('unspawnNotes', i, 'noteType') == 'Slice_Note' then --Check if the note on the chart is a Slice Note
			setPropertyFromGroup('unspawnNotes', i, 'texture', 'nevada/notes/Slice_Note'); --Change texture
			setPropertyFromGroup('unspawnNotes', i, 'noteSplashHue', 0); --custom notesplash color, why not
			setPropertyFromGroup('unspawnNotes', i, 'noteSplashSat', -20);
			setPropertyFromGroup('unspawnNotes', i, 'noteSplashBrt', 1);

			if getPropertyFromGroup('unspawnNotes', i, 'mustPress') then --Doesn't let Dad/Opponent notes get ignored
				setPropertyFromGroup('unspawnNotes', i, 'ignoreNote', false); --Miss has penalties
			end
		end
	end
end


local sliceAnims = {"slashLEFT"}
function goodNoteHit(id, direction, noteType, isSustainNote)
	if noteType == 'Slice_Note' then
		if difficulty == 1 then
			playSound('Slice', 0.5);
		elseif noteType == 'Slice_Note' and difficulty == 0 then
		       playSound('Slice', 0.5);
		end
		characterPlayAnim('dad', 'slashLEFT', false);
		characterPlayAnim('boyfriend', 'dodge', true);
		setProperty('boyfriend.specialAnim', true);
		setProperty('dad.specialAnim', true);
		cameraShake('camGame', 0.01, 0.2)
    end
end


local dodgeAnims = {"dodgeLEFT", "dodgeDOWN", "dodgeUP", "dodgeRIGHT"}
function goodNoteHit(id, direction, noteType, isSustainNote)
	if noteType == 'Slice_Note' then
	if difficulty == 1 then
			playSound('Slice', 0.5);
		elseif noteType == 'Slice_Note' and difficulty == 0 then
		       playSound('Slice', 0.5);
		end
		characterPlayAnim('dad', 'slashLEFT', false);
		characterPlayAnim('boyfriend', dodgeAnims[direction + 1], true);
		setProperty('boyfriend.specialAnim', true);
		cameraShake('camGame', 0.01, 0.2)
    end
end


function noteMiss(id, direction, noteType, isSustainNote)
	if noteType == 'Slice_Note' and difficulty == 1 then
		setProperty('health', -1);
		playSound('Slice', 0.5);
		playSound('Shanked', 0.6);
		playSound('D0nyDed', 0.6);
		characterPlayAnim('dad', 'slashLEFT', false);
	elseif noteType == 'Slice_Note' and difficulty == 0 then
		runTimer('bleed', 0.5, 60);
		playSound('D0nyDed', 0.7);
		playSound('Shanked', 0.7);
		playSound('Slice', 0.6);
		characterPlayAnim('dad', 'slashLEFT', true);
		characterPlayAnim('boyfriend', 'hurt', true);
	end
end`

**Comment by frantastic24** [comment]
there's no code triggering when the timer ends?

**Comment by FreakyOnAFriday** [comment]
You mean like this? 

`function onTimerCompleted(tag, loops, loopsLeft)
	-- A loop from a timer you called has been completed, value "tag" is it's tag
	-- loops = how many loops it will have done when it ends completely
	-- loopsLeft = how many are remaining
	if loopsLeft >= 1 then
		setProperty('health', getProperty('health')-0.001);
	end
end`

**Comment by frantastic24** [comment]
don't know why loopsLeft is being used but sure that works

**Comment by FreakyOnAFriday** [comment]
> don't know why loopsLeft is being used but sure that works

Either way the issue remains unfixed. The code will work on one song, and I will lose health as intended. But when it is ran on the two other songs I have tried it on it doesn't work correctly and I lose no health. Even before I added the timer it still worked as intended on the first song, but not on the last two for some reason?? Extremely confusing. I think it may be a Psych bug but I'm not sure. I know the lua is very experimental and all.

**Comment by FreakyOnAFriday** [comment]
I have no global scripts or anything else running btw.

**Comment by frantastic24** [comment]
are you on the right difficulty

**Comment by FreakyOnAFriday** [comment]
I am. I honestly think the JSON files for those two songs might be the problem because now I just tried it on bopeebo and another test song and it worked just fine. Sadly it's looking like I'm going to have to rechart them both.

**Comment by frantastic24** [comment]
the only thing that could cause this is if you aren't playing on the very first difficulty or the note type in the chart is somehow incorrect

**Comment by FreakyOnAFriday** [comment]
Yo it's the next day and I finally fixed it. I had to make a whole new blank Json file, tested the note on it, it worked, and then I copied the notes from the old Json file (the one that wasn't working) over to the new one and it now works like a charm. Still really weird because I'm not sure why that was even happening but I guess it got corrupted or bugged or something? Anyways it's fixed now so I guess I'll close this.
